freqgen/analytics.py

```diff
@@ -0,0 +1,107 @@
+from sqlite3 import Connection, connect
+from fastapi import Request
+
+db_location: str = "db/freq_analytics.sqlite"
+
+
+def check_and_create_db(db_location: str = db_location) -> Connection:
+    """Checks if the database exists. If the database does not exist, it will be created.
+    Arguments:
+        db_location: The location of the SQLite database file.
+    Returns:
+        sqlite3.Connection: A connection object for the SQLite database.
+    """
+
+    connection = connect(db_location)
+    cursor = connection.cursor()
+
+    check_query: str = """ SELECT name 
+            FROM sqlite_master 
+            WHERE type='table' 
+            AND name='analytics';"""
+
+    if cursor.execute(check_query).fetchone() is None:
+        create_query: str = """CREATE TABLE analytics (
+            id INTEGER PRIMARY KEY AUTOINCREMENT,
+            timestamp DATETIME DEFAULT CURRENT_TIMESTAMP,
+            ip_address TEXT,
+            user_agent TEXT,
+            method TEXT,
+            path TEXT,
+            best_station TEXT,
+            station_name TEXT,
+            verbatims TEXT,
+            tags TEXT,
+            artists TEXT
+        );"""
+        cursor.execute(create_query)
+        connection.commit()
+
+    return connection
+
+
+def log_analytics(
+    request: Request,
+    best_station: str,
+    station_name: str,
+    verbatims: list[str],
+    tags: list[str],
+    artists: list[str],
+    db_location: str = db_location,
+) -> None:
+    """Logs analytics data to the SQLite database.
+    Arguments:
+        request: The FastAPI request object.
+        best_station: The best station determined by the model.
+        station_name: The name of the generated station.
+        verbatims: The list of verbatims associated with the request.
+        tags: The list of tags associated with the request.
+        artists: The list of artists associated with the request.
+        db_location: The location of the SQLite database file.
+    """
+    connection = check_and_create_db(db_location)
+    cursor = connection.cursor()
+
+    insert_query = """INSERT INTO analytics 
+        (ip_address, user_agent, method, path, best_station, station_name, verbatims, tags, artists) 
+        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?);"""
+
+    data: tuple = (
+        request.client.host if request.client else None,
+        request.headers.get("user-agent"),
```

Comment:
J'avais pas capté mais on a pas le droit d'enregistrer l'IP d'un·e utilisateur·ice sans lui dire, c'est une donnée personnelle : 

```suggestion
    data: tuple = (
        request.headers.get("user-agent"),
```